packages/flutter/lib/src/widgets/editable.dart

```diff
@@ -206,16 +208,15 @@ class RawInputLine extends Scrollable {
   /// The color to use when painting the cursor.
   final Color cursorColor;
 
+  /// True if the text should wrap and span multiple lines, false if it should
+  /// stay on a single line and scroll when overflowed.
+  final bool multiline;
+
   /// The color to use when painting the selection.
   final Color selectionColor;
 
-  /// Optional builder function for a widget that controls the boundary of a
-  /// text selection.
-  final TextSelectionHandleBuilder selectionHandleBuilder;
-
-  /// Optional builder function for a set of controls for working with a
-  /// text selection (e.g. copy and paste).
-  final TextSelectionToolbarBuilder selectionToolbarBuilder;
+  /// Optional delegate for building the text selection handles and toolbar.
+  final TextSelectionUi selectionUi;
```

Comment:
We already have a TextSelectionDelegate that is used by the material layer to manipulate the selected text. How about TextSelectionControls?